File: JAVA prog/InsertionSort.java
import java.util.*;
public class InsertionSort {
    private static final Scanner sc = new Scanner(System.in);
    public static void main(String[]args) {
        int n = sc.nextInt();
        int arr [] = new int[n];
        for(int i=0; i<n; i++)
            arr[i] = sc.nextInt();
        InsertionSort(arr,n);
    }

    static  void InsertionSort (int [] arr, int n) {
        for(int i=1; i<n; i++) {
            int key = arr[i];
            int j = i-1;

            while(j>=0 && arr[j] > key) {
                arr[j+1] = arr[j];
                j--;
            }
            arr[j+1] = key;
        }
        for(int i=0; i<n; i++)
            System.out.print(arr[i]+" ");
    }
}

File: code/sorting/src/insertion_sort/insertion_sort.kt
import java.util.*
fun <T : Comparable<T>> insertionsort(items : MutableList<T>) : List<T>
{
    if (items.isEmpty())
    {
        return items
    }
    for (count in 1..items.count() - 1) 
    {
        val item = items[count]
        var i = count
        while (i>0 && item < items[i - 1])
        {
            items[i] = items[i - 1]
            i -= 1
        }
        items[i] = item
    }
    return items
}

fun main(args: Array<String>) 
{
    val names = mutableListOf("one", "Two", "Zack", "Daniel", "Adam")
    println(names)
    println(insertionsort(names))  
}
